src/Tests/dotnet-MsiInstallation.Tests/WorkloadSetTests.cs

```diff
@@ -213,37 +222,118 @@ public void UpdateToWorkloadSetVersionWithManifestsNotAvailable()
+        public void InstallWorkloadSetViaGlobalJson()
+        {
+            SetupWorkloadSetInGlobalJson(out var originalRollback);
 
-            GetRollback(directory).Should().NotBe(rollback);
+            VM.CreateRunCommand("dotnet", "workload", "install", "aspire")
+                .WithWorkingDirectory(SdkTestingDirectory)
+                .Execute().Should().Pass();
+
+            GetRollback(SdkTestingDirectory).Should().NotBe(originalRollback);
+        }
+
+        [Fact]
+        public void InstallWithGlobalJsonAndSkipManifestUpdate()
+        {
+            SetupWorkloadSetInGlobalJson(out var originalRollback);
+
+            VM.CreateRunCommand("dotnet", "workload", "install", "aspire", "--skip-manifest-update")
+                .WithWorkingDirectory(SdkTestingDirectory)
+                .Execute().Should().Fail()
+                .And.HaveStdErrContaining("--skip-manifest-update")
+                .And.HaveStdErrContaining(Path.Combine(SdkTestingDirectory, "global.json"));
+        }
+
+        [Fact]
+        public void InstallWithVersionAndSkipManifestUpdate()
+        {
+            InstallSdk();
+
+            VM.CreateRunCommand("dotnet", "workload", "install", "aspire", "--skip-manifest-update", "--version", "8.0.300-preview.0.24178.1")
+                .Execute().Should().Fail()
+                .And.HaveStdErrContaining("--skip-manifest-update")
+                .And.HaveStdErrContaining("--sdk-version");
+        }
+
+        [Fact]
+        public void UpdateShouldNotPinWorkloadSet()
+        {
+            InstallSdk();
+            UpdateAndSwitchToWorkloadSetMode(out _, out _);
+
+            AddNuGetSource(@"c:\SdkTesting\WorkloadSets");
+
+            //  Rename latest workload set so it won't be installed
+            VM.CreateRunCommand("cmd", "/c", "ren", @$"c:\SdkTesting\WorkloadSets\Microsoft.NET.Workloads.8.0.300-preview.*.24217.2.nupkg", $"Microsoft.NET.Workloads.8.0.300-preview.*.24217.2.bak")
+                .Execute().Should().Pass();
+
+            VM.CreateRunCommand("dotnet", "workload", "update")
+                .Execute().Should().Pass();
+
+            GetWorkloadVersion().Should().Be("8.0.300-preview.0.24178.1");
+
+            //  Bring latest workload set version back, so installing workload should update to it
+            VM.CreateRunCommand("cmd", "/c", "ren", @$"c:\SdkTesting\WorkloadSets\Microsoft.NET.Workloads.8.0.300-preview.*.24217.2.bak", $"Microsoft.NET.Workloads.8.0.300-preview.*.24217.2.nupkg")
+                .Execute().Should().Pass();
+
+            InstallWorkload("aspire", skipManifestUpdate: false);
+
+            GetWorkloadVersion().Should().Be("8.0.300-preview.0.24217.2");
+        }
+
+        [Fact]
+        public void WorkloadSetInstallationRecordIsWrittenCorrectly()
+        {
+            //  Should the workload set version or the package version be used in the registry?
+            throw new NotImplementedException();
+        }
+
+        [Fact]
```

Comment:
Should we have a test that tries to install from another feature band?

Reply:
I ended up testing this incidentally, as I switched to using 8.0.302 instead of an 8.0.300-preview.0 build.  That's how I found the issue with cross-band workload set GC.

Some of the VM tests are now failing though because of the band mismatch (because when you do a workload set update on an 8.0.300 band SDK, it won't install 8.0.300-preview.0 workload sets, which is what the tests are expecting).

I will probably try to sort through that and get all the tests working separately from this PR.